src/internal/connector/graph/concurrency_limiter_test.go

```diff
@@ -0,0 +1,70 @@
+package graph
+
+import (
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	khttp "github.com/microsoft/kiota-http-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/tester"
+)
+
+type ConcurrencyLimiterTestSuite struct {
+	tester.Suite
+}
+
+func TestConcurrencyLimiterSuite(t *testing.T) {
+	suite.Run(t, &ConcurrencyLimiterTestSuite{Suite: tester.NewUnitSuite(t)})
+}
+
+func (suite *ConcurrencyLimiterTestSuite) TestConcurrencyLimiterMiddleware() {
+	t := suite.T()
+	maxConcurrentRequests := 4
+
+	InitializeConcurrencyLimiter(maxConcurrentRequests)
```

Comment:
Since Initialize is a singleton, a pattern you might want to employ here instead is to move the internal func of `once.Do()` to its own func.  That way Initialize can call `once.Do(func() { cl = generateConcurrencyLimiter() })`, and this test can call `generateConcurrencyLimiter()` directly, without populating the global singleton.

Reply:
Ah! I see. I thought each test would be self contained and would get torn down on completion, thereby eliminating the singleton. 
But this is a great idea! I'll make this change